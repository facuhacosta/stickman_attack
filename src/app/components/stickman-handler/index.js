export { StickmanHandler } from './stickman-handler';