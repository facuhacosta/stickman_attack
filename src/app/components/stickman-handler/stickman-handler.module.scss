.stickman-handler {
  position: absolute;
  width: 100%;
  height: 70%;
  bottom: 0;
}