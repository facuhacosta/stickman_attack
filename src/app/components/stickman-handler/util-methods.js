const generatePosition = (enemies_proximity) => {
  return Math.floor(Math.random() * enemies_proximity) * 5;
};

const generateEnemyType = (enemies) => {
  const randomValue = Math.floor(Math.random() * 100);
  let flag = 0;
  let returnValue = ''
  Object.keys(enemies).some(key => {
    flag += enemies[key];
    if (flag > randomValue) {
      returnValue = key
      return key;
    };
  });
  return (returnValue);
};

const generateDelay = () => (Math.floor(Math.random() * 10));

export const generateWave = (enemies, enemies_proximity, wave_size) => {
  const wavePreset = [];
  for (let index = 0; index < wave_size; index++) {
    const enemyPreset = generateEnemyType(enemies);
    const positionPreset = generatePosition(enemies_proximity);
    const delayPreset = generateDelay();
    wavePreset.push({
      key: index,
      type: `${enemyPreset}`,
      position: `${positionPreset}`,
      delay: delayPreset,
      status: true
    });
  };
  return wavePreset;
}