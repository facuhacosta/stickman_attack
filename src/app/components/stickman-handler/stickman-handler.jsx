import React, { useState } from 'react';
import { StickMan } from '../stick-man/stick-man';
import style from './stickman-handler.module.scss';
import { generateWave } from './util-methods';
  
export function StickmanHandler({enemies_proximity, enemies, wave_size}) {

  const [wave, setWave] = useState(generateWave(enemies, enemies_proximity, wave_size));

  return (
    <div className={style['stickman-handler']}>
      {wave.map(enemy => 
        <StickMan
          key={enemy.key}
          enemy_type={enemy.type}
          damage_dealed={25}
          position={enemy.position}
          animation_delay={enemy.delay}
        />
      )}
    </div>
  );
}