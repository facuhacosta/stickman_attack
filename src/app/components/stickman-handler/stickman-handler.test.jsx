import React from 'react';
import { render } from '@testing-library/react';
import { StickmanHandler } from './stickman-handler';

describe('StickmanHandler Component', () => {

  // Matching Snapshots
  test('it should match the snapshot', () => {
    const { asFragment } = render(<StickmanHandler />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Unit tests

});