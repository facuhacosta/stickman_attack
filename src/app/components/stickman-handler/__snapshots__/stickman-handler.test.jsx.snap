// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickmanHandler Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
